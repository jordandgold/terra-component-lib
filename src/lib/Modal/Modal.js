import React from "react";
import Button from "../Button/Button";
import "./Modal.scss";

const Modal = props => {
  const { className, title, body, buttonOne, buttonTwo, closeModal } = props;
  return (
    <aside className={`modal ${className}`}>
      <button onClick={closeModal()}>close</button>
      <h4>{title}</h4>
      <p>{body}</p>
      {buttonOne && (
        <Button
          onClick={buttonOne.onClick}
          text={buttonOne.text}
          className={buttonOne.className}
        />
      )}
      {buttonTwo && (
        <Button
          onClick={buttonTwo.onClick}
          text={buttonTwo.text}
          className={buttonTwo.className}
        />
      )}
    </aside>
  );
};

export default Modal;
