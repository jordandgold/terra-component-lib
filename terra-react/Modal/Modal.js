import React from "react";
import Button from "../Button/Button";
import Icon from "../Icon/Icon";
import "./Modal.scss";

var Modal = function Modal(props) {
  var className = props.className,
      title = props.title,
      body = props.body,
      buttonOne = props.buttonOne,
      buttonTwo = props.buttonTwo,
      closeModal = props.closeModal,
      name = props.name;
  return React.createElement("aside", {
    className: "ter-modal ".concat(className)
  }, React.createElement("button", {
    className: "ter-modal__close",
    onClick: function onClick() {
      return closeModal(name);
    },
    "aria-label": "Close Modal"
  }, React.createElement(Icon, {
    type: "enclosed-x-dark-16px",
    size: "16px"
  })), React.createElement("div", {
    className: "ter-modal__body"
  }, React.createElement("h4", {
    className: "ter-modal__title"
  }, title), React.createElement("p", null, body), buttonOne && React.createElement(Button, {
    onClick: buttonOne.onClick,
    text: buttonOne.text,
    className: buttonOne.className
  }), buttonTwo && React.createElement(Button, {
    onClick: buttonTwo.onClick,
    text: buttonTwo.text,
    className: buttonTwo.className
  })));
};

export default Modal;