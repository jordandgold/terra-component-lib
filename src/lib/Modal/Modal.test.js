import React from "react";
import { shallow } from "enzyme";

import Modal from "./Modal";

describe("Modal", () => {
  let wrapper;
  let mockOnClickOne;
  let mockOnClickTwo;
  let mockCloseModal;
  let buttonOne;
  let buttonTwo;

  beforeEach(() => {
    mockOnClickOne = jest.fn();
    mockOnClickTwo = jest.fn();
    mockCloseModal = jest.fn();
    buttonOne = {
      onClick: mockOnClickOne,
      text: "button one mock text",
      className: "mock class",
      name: "hello"
    };
    buttonTwo = {
      onClick: mockOnClickTwo,
      text: "button one mock text",
      className: "mock class"
    };

    wrapper = shallow(
      <Modal
        name="hello"
        className="test"
        title="test title"
        body="test body"
        buttonOne={buttonOne}
        buttonTwo={buttonTwo}
        closeModal={mockCloseModal}
      />
    );
  });

  it("should match the snap shot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snap shot with one button", () => {
    wrapper = shallow(
      <Modal
        className="test"
        title="test title"
        body="test body"
        buttonOne={buttonOne}
        closeModal={mockCloseModal}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should match the snap shot with no buttons", () => {
    wrapper = shallow(
      <Modal
        className="test"
        title="test title"
        body="test body"
        closeModal={mockCloseModal}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should call close modal on click", () => {
    wrapper.find(".ter-modal__close").simulate("click");

    expect(mockCloseModal).toHaveBeenCalledWith("hello");
  });
});
