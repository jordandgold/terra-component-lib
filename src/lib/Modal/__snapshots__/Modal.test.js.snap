// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match the snap shot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    body="test body"
    buttonOne={
      Object {
        "className": "mock class",
        "onClick": [MockFunction],
        "text": "button one mock text",
      }
    }
    buttonTwo={
      Object {
        "className": "mock class",
        "onClick": [MockFunction],
        "text": "button one mock text",
      }
    }
    className="test"
    closeModal={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="ter-modal__close"
        >
          close
        </button>,
        <div
          className="ter-modal__body"
        >
          <h4
            class="ter-modal__title"
          >
            test title
          </h4>
          <p>
            test body
          </p>
          <Button
            className="mock class"
            onClick={[MockFunction]}
            text="button one mock text"
          />
          <Button
            className="mock class"
            onClick={[MockFunction]}
            text="button one mock text"
          />
        </div>,
      ],
      "className": "ter-modal test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "ter-modal__close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "close",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              class="ter-modal__title"
            >
              test title
            </h4>,
            <p>
              test body
            </p>,
            <Button
              className="mock class"
              onClick={[MockFunction]}
              text="button one mock text"
            />,
            <Button
              className="mock class"
              onClick={[MockFunction]}
              text="button one mock text"
            />,
          ],
          "className": "ter-modal__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test title",
              "class": "ter-modal__title",
            },
            "ref": null,
            "rendered": "test title",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test body",
            },
            "ref": null,
            "rendered": "test body",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "mock class",
              "onClick": [MockFunction],
              "text": "button one mock text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "mock class",
              "onClick": [MockFunction],
              "text": "button one mock text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="ter-modal__close"
          >
            close
          </button>,
          <div
            className="ter-modal__body"
          >
            <h4
              class="ter-modal__title"
            >
              test title
            </h4>
            <p>
              test body
            </p>
            <Button
              className="mock class"
              onClick={[MockFunction]}
              text="button one mock text"
            />
            <Button
              className="mock class"
              onClick={[MockFunction]}
              text="button one mock text"
            />
          </div>,
        ],
        "className": "ter-modal test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "ter-modal__close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                class="ter-modal__title"
              >
                test title
              </h4>,
              <p>
                test body
              </p>,
              <Button
                className="mock class"
                onClick={[MockFunction]}
                text="button one mock text"
              />,
              <Button
                className="mock class"
                onClick={[MockFunction]}
                text="button one mock text"
              />,
            ],
            "className": "ter-modal__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test title",
                "class": "ter-modal__title",
              },
              "ref": null,
              "rendered": "test title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test body",
              },
              "ref": null,
              "rendered": "test body",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "mock class",
                "onClick": [MockFunction],
                "text": "button one mock text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "mock class",
                "onClick": [MockFunction],
                "text": "button one mock text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Modal should match the snap shot with no buttons 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    body="test body"
    className="test"
    closeModal={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="ter-modal__close"
        >
          close
        </button>,
        <div
          className="ter-modal__body"
        >
          <h4
            class="ter-modal__title"
          >
            test title
          </h4>
          <p>
            test body
          </p>
        </div>,
      ],
      "className": "ter-modal test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "ter-modal__close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "close",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              class="ter-modal__title"
            >
              test title
            </h4>,
            <p>
              test body
            </p>,
            undefined,
            undefined,
          ],
          "className": "ter-modal__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test title",
              "class": "ter-modal__title",
            },
            "ref": null,
            "rendered": "test title",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test body",
            },
            "ref": null,
            "rendered": "test body",
            "type": "p",
          },
          undefined,
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="ter-modal__close"
          >
            close
          </button>,
          <div
            className="ter-modal__body"
          >
            <h4
              class="ter-modal__title"
            >
              test title
            </h4>
            <p>
              test body
            </p>
          </div>,
        ],
        "className": "ter-modal test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "ter-modal__close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                class="ter-modal__title"
              >
                test title
              </h4>,
              <p>
                test body
              </p>,
              undefined,
              undefined,
            ],
            "className": "ter-modal__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test title",
                "class": "ter-modal__title",
              },
              "ref": null,
              "rendered": "test title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test body",
              },
              "ref": null,
              "rendered": "test body",
              "type": "p",
            },
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Modal should match the snap shot with one button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    body="test body"
    buttonOne={
      Object {
        "className": "mock class",
        "onClick": [MockFunction],
        "text": "button one mock text",
      }
    }
    className="test"
    closeModal={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="ter-modal__close"
        >
          close
        </button>,
        <div
          className="ter-modal__body"
        >
          <h4
            class="ter-modal__title"
          >
            test title
          </h4>
          <p>
            test body
          </p>
          <Button
            className="mock class"
            onClick={[MockFunction]}
            text="button one mock text"
          />
        </div>,
      ],
      "className": "ter-modal test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "ter-modal__close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "close",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              class="ter-modal__title"
            >
              test title
            </h4>,
            <p>
              test body
            </p>,
            <Button
              className="mock class"
              onClick={[MockFunction]}
              text="button one mock text"
            />,
            undefined,
          ],
          "className": "ter-modal__body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test title",
              "class": "ter-modal__title",
            },
            "ref": null,
            "rendered": "test title",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "test body",
            },
            "ref": null,
            "rendered": "test body",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "mock class",
              "onClick": [MockFunction],
              "text": "button one mock text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="ter-modal__close"
          >
            close
          </button>,
          <div
            className="ter-modal__body"
          >
            <h4
              class="ter-modal__title"
            >
              test title
            </h4>
            <p>
              test body
            </p>
            <Button
              className="mock class"
              onClick={[MockFunction]}
              text="button one mock text"
            />
          </div>,
        ],
        "className": "ter-modal test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "ter-modal__close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                class="ter-modal__title"
              >
                test title
              </h4>,
              <p>
                test body
              </p>,
              <Button
                className="mock class"
                onClick={[MockFunction]}
                text="button one mock text"
              />,
              undefined,
            ],
            "className": "ter-modal__body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test title",
                "class": "ter-modal__title",
              },
              "ref": null,
              "rendered": "test title",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test body",
              },
              "ref": null,
              "rendered": "test body",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "mock class",
                "onClick": [MockFunction],
                "text": "button one mock text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
